### PR TITLE
test: deflaking unexpected disconnects permanently

### LINKED ISSUE #12861: Test100AndDisconnectLegacy flake (ARM release)
```
[ RUN      ] Protocols/DownstreamProtocolIntegrationTest.Test100AndDisconnectLegacy/IPv4_HttpDownstream_HttpUpstream
[2020-08-27 21:11:58.717][30877][critical][assert] [test/integration/http_integration.cc:317] assert failure: result. Details: The connection disconnected unexpectedly, and allow_unexpected_disconnects_ is false.
```

### PR DESCRIPTION
**Pull request by alyssawilk**
Allowing unexpected disconnects by default.

Risk Level: Low
Testing: tests pass
Docs Changes: n/a
Release Notes: n/a
Fixes #12861

### CODE REVIEW

File: test/integration/fake_upstream.cc [hunk 1]
```diff
@@ -447,7 +447,7 @@ FakeUpstream::FakeUpstream(Network::TransportSocketFactoryPtr&& transport_socket
       api_(Api::createApiForTest(stats_store_)), time_system_(time_system),
       dispatcher_(api_->allocateDispatcher("fake_upstream")),
       handler_(new Server::ConnectionHandlerImpl(*dispatcher_)),
-      allow_unexpected_disconnects_(false), read_disable_on_new_connection_(true),
+      allow_unexpected_disconnects_(true), read_disable_on_new_connection_(true),
```
Comment (alyssawilk) [inline test/integration/fake_upstream.cc:450]: really my only question is if we should remove allow_unexpected_disconnects_ and set_allow_unexpected_disconnects() entirely, which will likely break downstream test builds, but it's only worth keeping around if someone really wants some test to allow_unexpected_disconnects_(false) explicitly.  Maybe a follow-up so we can roll only that part back if someone wants that?

Comment (snowp) [reply test/integration/fake_upstream.cc:450]: I think doing it in a follow up makes sense. Do we not have any tests that want to test for unexpected disconnects right now?

Comment (alyssawilk) [reply test/integration/fake_upstream.cc:450]: not that I know of?  I mean we could arguably set it false for every test which didn't previously have it set true, but I think most tests which wanted to know they got a complete request waited for a complete requests, ditto response.

Comment (snowp) [reply test/integration/fake_upstream.cc:450]: I don't think that's necessary, probably fine to just get rid of for now unless someone complains

Comment (mattklein123) [reply test/integration/fake_upstream.cc:450]: I'm fine with a follow up but I would be fine just getting rid of it all now.

**Comment by alyssawilk** [comment]
went with Matt's request and did it in one fell swoop.  LGTY?

File: test/integration/fake_upstream.h [hunk 2]
```diff
@@ -343,13 +333,8 @@ class SharedConnectionWrapper : public Network::ConnectionCallbacks,
...
-    if (unexpected_disconnect && !allow_unexpected_disconnects_) {
-      return testing::AssertionFailure()
-             << "The connection disconnected unexpectedly, and allow_unexpected_disconnects_ is "
-                "false."
-                "\n See "
-                "https://github.com/envoyproxy/envoy/blob/master/test/integration/README.md#"
-                "unexpected-disconnects";
+    if (unexpected_disconnect && !allow_disconnects) {
+      ENVOY_LOG_MISC(warn, "executeOnDispatcher failed due to disconnect\n");
     }
```
Comment (mattklein123) [inline test/integration/fake_upstream.h:338]: Is anyone ever going to see this? Delete also?

**Review by mattklein123 — COMMENTED** [review]
Awesome, just one comment for your thoughts on.

/wait-any

File: test/integration/fake_upstream.h (hunk 2 above)
Comment (alyssawilk) [reply test/integration/fake_upstream.h:338]: My thought, and why I left it at warn, was if some executeOnDispatcher call asserts the call happens when in fact it's an unexpected disconnect, it'd be a pain to debug without at least some breadcrumb.

Comment (mattklein123) [reply test/integration/fake_upstream.h:338]: OK SGTM. We can always iterate further later!

**Review by mattklein123 — APPROVED** [review]
Yay!

**Review by snowp — APPROVED** [review]
Nice, thanks!

**Pull request merged**